WcaOnRails/app/models/competition.rb

```diff
@@ -824,6 +824,25 @@ def to_wcif
     }
   end
 
+  def set_wcif_events!(wcif_events)
+    # Remove extra events.
```

Comment:
That makes sense!

So when you do `competition_event.destroy!` here the foreign keys are taken care of?

I think it's really important to write code for the schema you have, not for every schema you might possibly have in the future. This cuts down your workload a lot and makes fast progress possible.

If someone ever does add foreign keys to rounds, they'd have to make sure it works with all code that deals with rounds. Well written tests today can help with that, but you can't cover all possible future changes.